src/pyspiffe/workloadapi/default_x509_source.py

```diff
@@ -0,0 +1,129 @@
+
+           It blocks until the initial update has been received from the Workload API or until timeout_in_seconds is reached.
+
+           In case the underlying Workload API connection returns an unretryable error, the source will be closed and
+           no methods on the source will be available.
+
+
+        Args:
+            workload_api_client: A WorkloadApiClient that will be used to fetch the X.509 materials from the Workload API.
+                                 In case it's not provided, a default client will be created.
+
+            spiffe_socket: Path to Workload API UDS. This will be used in case a the workload_api_client is not provided.
+                           If not specified, the SPIFFE_ENDPOINT_SOCKET environment variable must be set.
+
+            timeout_in_seconds: Time to wait for the first update of the Workload API. If no timeout is provided, and
+                                the connection with the Workload API fails, it will block Indefinitely while
+                                the connection is retried.
+
+            picker: Function to choose the X.509 SVID from the list returned by the Workload API.
+                    If it is not set, the default SVID is picked. If the picker function throws an error,
+                    it will render the X509Source invalid and it will be closed.
+
+        Returns:
+            DefaultX509Source: New DefaultX509Source object, initialized with the X509Context fetched from the Workload API.
+
+        Raises:
+            ArgumentError: If spiffe_socket_path is invalid or not provided and SPIFFE_ENDPOINT_SOCKET env variable is not set.
+
+            X509SourceError: In case a timeout was configured and it was reached during the source initialization waiting
+                             for the first update from the Workload API.
+        """
+
+        self._initialized = threading.Event()
+        self._lock = threading.Lock()
+        self._closed = False
+        self._workload_api_client = (
+            workload_api_client
+            if workload_api_client
+            else DefaultWorkloadApiClient(spiffe_socket)
+        )
+        self._picker = picker
+
+        # set the watcher that will keep the source updated and log the underlying errors
+        self._client_cancel_handler = self._workload_api_client.watch_x509_context(
+            self._set_context, self._log_error
+        )
+
+        self._initialized.wait(timeout_in_seconds)
+
+        if not self._initialized.is_set():
+            self._client_cancel_handler.cancel()
+            raise X509SourceError(
+                'Could not initialize X.509 Source: reached timeout waiting for the first update'
+            )
+
+    def get_x509_svid(self) -> X509Svid:
+        """Returns an X509-SVID from the source."""
+        with self._lock:
+            if self._closed:
+                raise X509SourceError('Cannot get X.509 SVID: source is closed')
+        return self._x509_svid
```

Comment:
access to x509_svid should be thread safe too, this should be inside the with block.